gui/src/main/java/org/verapdf/cli/VeraPdfCliProcessor.java

```diff
@@ -3,90 +3,70 @@
-import org.verapdf.pdfa.results.TestAssertion;
-import org.verapdf.pdfa.results.TestAssertion.Status;
-import org.verapdf.pdfa.results.ValidationResult;
-import org.verapdf.pdfa.validation.Profiles;
-import org.verapdf.pdfa.validation.RuleId;
-import org.verapdf.pdfa.validation.ValidationProfile;
-import org.verapdf.pdfa.validators.Validators;
+import org.verapdf.processor.config.Config;
 import org.verapdf.report.*;
 
 import javax.xml.bind.JAXBException;
-import javax.xml.transform.TransformerException;
 import java.io.*;
-import java.nio.file.FileAlreadyExistsException;
 import java.nio.file.FileSystems;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 /**
  * @author <a href="mailto:carl@openpreservation.org">Carl Wilson</a>
  *
  */
 final class VeraPdfCliProcessor {
-    final FormatOption format;
-    final ProcessingType processingType;
-    final boolean logPassed;
-    final boolean recurse;
-    final boolean verbose;
-    final PDFAValidator validator;
-
-    final int maxFailuresDisplayed;
-
-    final boolean fixMetadata;
-    final String prefix;
-    final Path saveFolder;
 
-    final String profilesWikiPath;
+	private static final Logger LOGGER = Logger.getLogger(VeraPdfCliProcessor.class);
 
-    final ValidationProfile profile;
+    final boolean recurse;
 
-    private String currentPdfName;
+    private Config config;
 
     private VeraPdfCliProcessor() throws IOException {
         this(new VeraCliArgParser());
     }
 
+	public Config getConfig() {
+		return this.config;
+	}
+
     private VeraPdfCliProcessor(final VeraCliArgParser args)
             throws IOException {
-        this.format = args.getFormat();
-        this.processingType = args.getProcessingType();
-        this.logPassed = args.logPassed();
-        this.recurse = args.isRecurse();
-        this.verbose = args.isVerbose();
-        this.profile = profileFromArgs(args);
-        this.validator = (this.profile == Profiles.defaultProfile()) ? null
-                : Validators.createValidator(this.profile, logPassed(args), args.maxFailures());
-
-        this.maxFailuresDisplayed = args.maxFailuresDisplayed();
-
-        this.fixMetadata = args.fixMetadata();
-        this.prefix = args.prefix();
-        this.saveFolder = FileSystems.getDefault().getPath(args.saveFolder());
-        this.profilesWikiPath = args.getProfilesWikiPath();
-    }
-
-    private static boolean logPassed(final VeraCliArgParser args) {
-        return (args.getFormat() != FormatOption.XML) || args.logPassed();
+		this.recurse = args.isRecurse();
+
+
+		if (args.isLoadingConfig()) {
+			try {
+				config = ConfigIO.readConfig();
+			} catch (IOException e) {
+				LOGGER.error("Can not read config file", e);
+				this.config = new Config();
+			} catch (JAXBException e) {
+				LOGGER.error("Cannot parse config XML", e);
+				this.config = new Config();
+			}
+		} else {
```

Comment:
if we are loading config and have some another settings, then they should override config values